chat/consumers.py

```diff
@@ -0,0 +1,43 @@
+import json
+from channels.generic.websocket import AsyncWebsocketConsumer
+
+class ChatConsumer(AsyncWebsocketConsumer):
+    """
+    WebSocket consumer for handling real-time chat messages in specific rooms.
+    """
+
+    async def connect(self):
+        """
+        Handles a new WebSocket connection.
+        Joins the corresponding room group based on the room name.
+        """
+        self.room_name = self.scope["url_route"]["kwargs"]["room_name"]
+        self.room_group_name = f"chat_{self.room_name}"
+        await self.channel_layer.group_add(self.room_group_name, self.channel_name)
```

Comment:
You may want to wrap group_add in a try/except block and log any exceptions to handle channel layer errors gracefully.